src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerActionCounts.cs

```diff
@@ -73,7 +73,8 @@ internal class AnalyzerActionCounts
                 SyntaxNodeActionsCount > 0 ||
                 OperationActionsCount > 0 ||
                 OperationBlockActionsCount > 0 ||
-                OperationBlockStartActionsCount > 0;
+                OperationBlockStartActionsCount > 0 ||
+                SymbolStartActionsCount > 0;
```

Comment:
Single line change in compiler code: `HasAnyExecutableCodeActions` conservatively returns true for analyzers with SymbolStartActions as we cannot compute this upfront for an analyzer (need to actually execute the symbol start action for each symbol).